backend/app/cmd/server.go

```diff
@@ -347,7 +358,15 @@ func (s *ServerCommand) newServerApp() (*serverApp, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to make pictures store")
 	}
-	log.Printf("[DEBUG] image service for url=%s, EditDuration=%v", imageService.ImageAPI, imageService.EditDuration)
+
+	searchService, err := s.makeSearchService()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to make search service")
+	}
+
+	if searchService != nil {
+		storeEngine = search.WrapEngine(storeEngine, searchService)
```

Comment:
This doesn't feel right, I'll write if I'll have any idea on how to make it better.
I think either storeEngine should already have search embedded 100% of the time with something like NoopSearchService if it's not present, or search should be a separate object with storeEngine passed to it explicitly and unaltered.

@umputun what do you think about this particular place?

Reply:
I think that with NoopSearchService it should be possible to do what you do in the wrapper in the normal Engine, even for a remote one. But then we'll the same logic in multiple places which is bad as well.

Reply:
Should I wrap storeEngine with NoopSearchService ? How I should get wrappedEngine in such flow? Or I may try other approach without wrapping?

Reply:
I'll think about it. But wrapping shouldn't be conditional, there should be either real or Noop search engine and the flow should be the same for both.

Reply:
I've create NoOp service but sore.Interface still wrapped. Proxy used in all cases whether search enabled or not (noop used)